libs/blocks/video/video.js

```diff
@@ -22,4 +24,17 @@ export default function init(a) {
   const videoElem = document.body.querySelector(`source[src="${videoPath}"]`)?.parentElement;
   applyHoverPlay(videoElem);
   a.remove();
+};
+
+export default function init(a) {
+  a.classList.add('hide');
+  if (a.textContent.includes('no-lazy')) {
+    loadVideo(a);
+  } else {
+    createIntersectionObserver({
```

Comment:
ideally there should be 1 intersection observer created for all the videos not one for each video.

Reply:
That is kinda difficult approach with current milo structure.
Maybe we can come up a solution adding option at autoblock stage and get the logic there, but probably not with this PR.
https://github.com/adobecom/milo/blob/main/libs/utils/utils.js#L76

Reply:
I addressed the video issue as at https://github.com/adobecom/milo/blob/perfbs/libs/blocks/video/video.js, not sure which one is more convenient.